File: Examples/Chart/WinForm/CSharp/NMainForm.cs
using System;
using System.Drawing;
using System.Windows.Forms;
using Nevron.Chart.WinForm;
using Nevron.Examples.Framework.WinForm;
using Nevron.UI.WinForm.Docking;
using Nevron.UI.WinForm.Controls;

namespace Nevron.Examples.Chart.WinForm
{
	public class NMainForm : NExamplesForm
	{
		#region Fields

		internal NChartControl chartControl;
		internal NChartCommandBarsManager chartCommandBarsManager;
		internal ExampleLayout m_CurrentLayout = ExampleLayout.Standard;

		#endregion

		#region Constructors

		public NMainForm()
		{
			InitializeComponent();

			InitFromConfig(new NChartExamplesConfig());

			this.StartPosition = FormStartPosition.CenterScreen;
			this.WindowState = FormWindowState.Maximized;

			this.chartControl = new Nevron.Chart.WinForm.NChartControl();

			this.chartCommandBarsManager = new NChartCommandBarsManager();
			this.chartCommandBarsManager.ChartControl = chartControl;
			this.chartCommandBarsManager.ParentControl = this.MainControlHost;
			this.chartCommandBarsManager.AllowCustomize = false;
			this.chartCommandBarsManager.ContextMenuEnabled = true;
			SetupToolbarCommands();

			this.MainControlHost.Controls.Add(chartControl);

			this.chartControl.Dock = DockStyle.Fill;
			this.chartControl.BringToFront();

		}

		#endregion

		#region Windows Form Designer generated code
		/// <summary>
		/// Required method for Designer support - do not modify
		/// the contents of this method with the code editor.
		/// </summary>
		private void InitializeComponent()
		{
			this.Name = "MainForm";
			this.Text = "Nevron Chart for .NET - C# examples";
		}

		#endregion

		#region Main
		
		/// <summary>
		/// The main entry point for the application.
		/// </summary>
		[STAThread]
		static void Main()
		{
			NMainForm form = new NMainForm();

			Application.Run(form);
		}

		#endregion

		#region Overrides

		protected override void OnFormLoading()
		{
			base.OnFormLoading();

			this.DockManager.m_ControlHost.SizeInfo.SizeLogic = SizeLogic.FillInterior;
		}

		protected override void ClearExample(bool clearAll)
		{
			base.ClearExample(clearAll);

			this.chartControl.Clear();
			this.chartControl.ServiceManager.LegendUpdateService.Start();
			this.chartControl.DisposeEvents();
		}

		protected override void LoadExample(NExampleEntity example)
		{
			base.LoadExample(example);
			this.chartControl.Refresh();
		}

		protected override void LayoutExample()
		{
			NExampleBaseUC uc = CurrentExampleControl as NExampleBaseUC;

			if(uc == null)
				return;

			int nWidth = uc.Width + 10;
			int nHeight = uc.Height + 60;

			ExampleLayout newLayout = (nWidth > 330) ? ExampleLayout.WideScreen : ExampleLayout.Standard;

			if (m_CurrentLayout == newLayout)
			{
				if (ExamplePanel.ParentZone != null)
				{
					if (newLayout == ExampleLayout.WideScreen)
					{
						ExamplePanel.SizeInfo.AbsoluteSize = new Size(0, nHeight);
					}
					else if(newLayout == ExampleLayout.Standard)
					{
						ExamplePanel.SizeInfo.AbsoluteSize = new Size(nWidth, 0);
					}
				}
			}
			else
			{
				m_CurrentLayout = newLayout;

				NDockingPanelContainer mainContainer = DockManager.MainContainer;

				if(newLayout == ExampleLayout.WideScreen)
				{
					INDockZone zone = DockManager.m_ControlHost.ParentZone;
					NDockingPanelHost dpHost = FindDockingPanelHostInZone(zone);

					if(dpHost != null)
					{
						ExamplePanel.PerformDock(dpHost, DockStyle.Fill, 0);
					}
					else
					{
						ExamplePanel.PerformDock(zone, DockStyle.Bottom, 0);
					}

					ExamplePanel.SizeInfo.AbsoluteSize = new Size(0, nHeight);
				}
				else if(newLayout == ExampleLayout.Standard)
				{
					ExamplePanel.PerformDock(mainContainer.RootZone, DockStyle.Right, 0);
					ExamplePanel.SizeInfo.AbsoluteSize = new Size(nWidth, 0);
				}
			}
		}

		protected override NExampleUserControl CreateExampleControl(NExampleEntity example)
		{
			NExampleBaseUC uc = (NExampleBaseUC)base.CreateExampleControl(example);

			uc.nChartControl1 = chartControl;

			return uc;
		}

		#endregion

		#region Implementation

		private NDockingPanelHost FindDockingPanelHostInZone(INDockZone zone)
		{
			if(zone == null)
				return null;

			if(zone.LayoutInfo.Orientation != Orientation.Vertical)
				return null;

			foreach(INDockZoneChild child in zone.Children)
			{
				NDockingPanelHost host = child as NDockingPanelHost;

				if(host != null)
					return host;
			}

			return null;
		}
		private void SetupToolbarCommands()
		{
			return;

/*			chartCommandBarsManager.Toolbars["Aspect"].Visible = false;
			chartCommandBarsManager.Toolbars["Panel"].Visible = false;
			chartCommandBarsManager.Toolbars["Format"].Visible = false;

			NCommandCollection standardCommands = chartCommandBarsManager.Toolbars["Standard"].Commands;
			standardCommands.RemoveAt(7);
			standardCommands.RemoveAt(6);
			standardCommands.RemoveAt(5);
			standardCommands.RemoveAt(1);
			standardCommands.RemoveAt(0);

			NCommandCollection toolsCommands = chartCommandBarsManager.Toolbars["Tools"].Commands;
			toolsCommands.RemoveAt(3);
			toolsCommands.RemoveAt(2);

			NCommandCollection projectionCommands = chartCommandBarsManager.Toolbars["Projection"].Commands;
			projectionCommands.RemoveAt(12);
			projectionCommands.RemoveAt(8);
			projectionCommands.RemoveAt(7);
			projectionCommands.RemoveAt(6);
			projectionCommands.RemoveAt(5);
			projectionCommands.RemoveAt(4);
			projectionCommands.RemoveAt(3);
			projectionCommands.RemoveAt(2);
			projectionCommands.RemoveAt(1);

			for (int i = 0; i < chartCommandBarsManager.Toolbars.Count; i++)
			{
				NDockingToolbar toolbar = chartCommandBarsManager.Toolbars[i];
				toolbar.HasPendantCommand = false;
				toolbar.AllowHide = false;
				toolbar.AllowDelete = false;
				toolbar.RowIndex = 0;
			}*/
		}

		#endregion
	}
}

File: Examples/Chart/WinForm/VB.NET/NMainForm.vb
Imports Nevron.Chart.WinForm
Imports Nevron.Examples.Framework.WinForm
Imports Nevron.UI.WinForm.Docking

Namespace Nevron.Examples.Chart.WinForm
	Public Class NMainForm
		Inherits NExamplesForm

		#Region "Fields"

		Friend chartControl As NChartControl
		Friend chartCommandBarsManager As NChartCommandBarsManager
		Friend m_CurrentLayout As ExampleLayout = ExampleLayout.Standard

		#End Region

		#Region "Constructors"

		Public Sub New()
			InitializeComponent()

			InitFromConfig(New NChartExamplesConfig())

			Me.StartPosition = FormStartPosition.CenterScreen
			Me.WindowState = FormWindowState.Maximized

			Me.chartControl = New Nevron.Chart.WinForm.NChartControl()

			Me.chartCommandBarsManager = New NChartCommandBarsManager()
			Me.chartCommandBarsManager.ChartControl = chartControl
			Me.chartCommandBarsManager.ParentControl = Me.MainControlHost
			Me.chartCommandBarsManager.AllowCustomize = False
			Me.chartCommandBarsManager.ContextMenuEnabled = True
			SetupToolbarCommands()

			Me.MainControlHost.Controls.Add(chartControl)

			Me.chartControl.Dock = DockStyle.Fill
			Me.chartControl.BringToFront()

		End Sub

		#End Region

		#Region "Windows Form Designer generated code"
		''' <summary>
		''' Required method for Designer support - do not modify
		''' the contents of this method with the code editor.
		''' </summary>
		Private Sub InitializeComponent()
			Me.Name = "MainForm"
			Me.Text = "Nevron Chart for .NET - VB.NET examples"
		End Sub

		#End Region

		#Region "Main"

		''' <summary>
		''' The main entry point for the application.
		''' </summary>
		<STAThread>
		Shared Sub Main()
			Dim form As New NMainForm()

			Application.Run(form)
		End Sub

		#End Region

		#Region "Overrides"

		Protected Overrides Sub OnFormLoading()
			MyBase.OnFormLoading()

			Me.DockManager.m_ControlHost.SizeInfo.SizeLogic = SizeLogic.FillInterior
		End Sub

		Protected Overrides Sub ClearExample(ByVal clearAll As Boolean)
			MyBase.ClearExample(clearAll)

			Me.chartControl.Clear()
			Me.chartControl.ServiceManager.LegendUpdateService.Start()
			Me.chartControl.DisposeEvents()
		End Sub

		Protected Overrides Sub LoadExample(ByVal example As NExampleEntity)
			MyBase.LoadExample(example)
			Me.chartControl.Refresh()
		End Sub

		Protected Overrides Sub LayoutExample()
			Dim uc As NExampleBaseUC = TryCast(CurrentExampleControl, NExampleBaseUC)

			If uc Is Nothing Then
				Return
			End If

			Dim nWidth As Integer = uc.Width + 10
			Dim nHeight As Integer = uc.Height + 60

			Dim newLayout As ExampleLayout = If(nWidth > 330, ExampleLayout.WideScreen, ExampleLayout.Standard)

			If m_CurrentLayout = newLayout Then
				If ExamplePanel.ParentZone IsNot Nothing Then
					If newLayout = ExampleLayout.WideScreen Then
						ExamplePanel.SizeInfo.AbsoluteSize = New Size(0, nHeight)
					ElseIf newLayout = ExampleLayout.Standard Then
						ExamplePanel.SizeInfo.AbsoluteSize = New Size(nWidth, 0)
					End If
				End If
			Else
				m_CurrentLayout = newLayout

				Dim mainContainer As NDockingPanelContainer = DockManager.MainContainer

				If newLayout = ExampleLayout.WideScreen Then
					Dim zone As INDockZone = DockManager.m_ControlHost.ParentZone
					Dim dpHost As NDockingPanelHost = FindDockingPanelHostInZone(zone)

					If dpHost IsNot Nothing Then
						ExamplePanel.PerformDock(dpHost, DockStyle.Fill, 0)
					Else
						ExamplePanel.PerformDock(zone, DockStyle.Bottom, 0)
					End If

					ExamplePanel.SizeInfo.AbsoluteSize = New Size(0, nHeight)
				ElseIf newLayout = ExampleLayout.Standard Then
					ExamplePanel.PerformDock(mainContainer.RootZone, DockStyle.Right, 0)
					ExamplePanel.SizeInfo.AbsoluteSize = New Size(nWidth, 0)
				End If
			End If
		End Sub

		Protected Overrides Function CreateExampleControl(ByVal example As NExampleEntity) As NExampleUserControl
			Dim uc As NExampleBaseUC = DirectCast(MyBase.CreateExampleControl(example), NExampleBaseUC)

			uc.nChartControl1 = chartControl

			Return uc
		End Function

		#End Region

		#Region "Implementation"

		Private Function FindDockingPanelHostInZone(ByVal zone As INDockZone) As NDockingPanelHost
			If zone Is Nothing Then
				Return Nothing
			End If

			If zone.LayoutInfo.Orientation <> Orientation.Vertical Then
				Return Nothing
			End If

			For Each child As INDockZoneChild In zone.Children
				Dim host As NDockingPanelHost = TryCast(child, NDockingPanelHost)

				If host IsNot Nothing Then
					Return host
				End If
			Next child

			Return Nothing
		End Function
		Private Sub SetupToolbarCommands()
			Return

'			chartCommandBarsManager.Toolbars["Aspect"].Visible = false;
'			chartCommandBarsManager.Toolbars["Panel"].Visible = false;
'			chartCommandBarsManager.Toolbars["Format"].Visible = false;
'
'			NCommandCollection standardCommands = chartCommandBarsManager.Toolbars["Standard"].Commands;
'			standardCommands.RemoveAt(7);
'			standardCommands.RemoveAt(6);
'			standardCommands.RemoveAt(5);
'			standardCommands.RemoveAt(1);
'			standardCommands.RemoveAt(0);
'
'			NCommandCollection toolsCommands = chartCommandBarsManager.Toolbars["Tools"].Commands;
'			toolsCommands.RemoveAt(3);
'			toolsCommands.RemoveAt(2);
'
'			NCommandCollection projectionCommands = chartCommandBarsManager.Toolbars["Projection"].Commands;
'			projectionCommands.RemoveAt(12);
'			projectionCommands.RemoveAt(8);
'			projectionCommands.RemoveAt(7);
'			projectionCommands.RemoveAt(6);
'			projectionCommands.RemoveAt(5);
'			projectionCommands.RemoveAt(4);
'			projectionCommands.RemoveAt(3);
'			projectionCommands.RemoveAt(2);
'			projectionCommands.RemoveAt(1);
'
'			for (int i = 0; i < chartCommandBarsManager.Toolbars.Count; i++)
'			{
'				NDockingToolbar toolbar = chartCommandBarsManager.Toolbars[i];
'				toolbar.HasPendantCommand = false;
'				toolbar.AllowHide = false;
'				toolbar.AllowDelete = false;
'				toolbar.RowIndex = 0;
'			}
		End Sub

		#End Region
	End Class
End Namespace